azurelinuxagent/common/exception.py

```diff
@@ -100,6 +100,19 @@ def __init__(self, msg=None, inner=None, code=-1):
         super(ExtensionDownloadError, self).__init__(msg, inner, code)
 
 
+class ExtensionStatusError(ExtensionError):
+    """
+    When extension failed to provide a valid status file
+    """
+    Unknown = -1
```

Comment:
What is the use of Unknown? Currently I don't see it being used anywhere. You may want to force the  caller to provide a value.

Reply:
Also, I think we need to start defining those exceptions on the module that raises them. They are part of the module's interface.

Reply:
> What is the use of Unknown? Currently I don't see it being used anywhere. You may want to force the caller to provide a value.

Yes. Unknown was useless - Removed it now and made code compulsory.

> Also, I think we need to start defining those exceptions on the module that raises them. They are part of the module's interface.

I partially agree with this. I don't want to break away the current exceptions in this PR and it is better to see all the exceptions together.